syntax = "proto3";

package worker.v1;

option go_package = "github.com/superblocksteam/agent/types/gen/go/worker/v1";

service SandboxVariableStoreService {
  rpc GetVariable(GetVariableRequest) returns (GetVariableResponse);
  rpc SetVariable(SetVariableRequest) returns (SetVariableResponse);
  rpc GetVariables(GetVariablesRequest) returns (GetVariablesResponse);
  rpc SetVariables(SetVariablesRequest) returns (SetVariablesResponse);
  // FetchFile fetches file contents from the orchestrator's file server.
  // The task-manager handles authentication with the orchestrator.
  rpc FetchFile(FetchFileRequest) returns (FetchFileResponse);
}

// Variable operations
message GetVariableRequest {
  string execution_id = 1;
  string key = 2;
}

message GetVariableResponse {
  string value = 1;  // JSON-encoded value
  bool found = 2;
}

message SetVariableRequest {
  string execution_id = 1;
  string key = 2;
  string value = 3;  // JSON-encoded value
}

message SetVariableResponse {
  bool success = 1;
}

message GetVariablesRequest {
  string execution_id = 1;
  repeated string keys = 2;
}

message GetVariablesResponse {
  repeated string values = 1;  // JSON-encoded values (same order as keys)
}

message SetVariablesRequest {
  string execution_id = 1;
  repeated KeyValue kvs = 2;
}

message KeyValue {
  string key = 1;
  string value = 2;  // JSON-encoded value
}

message SetVariablesResponse {
  bool success = 1;
}

// File operations - sandbox requests files from task-manager
message FetchFileRequest {
  string execution_id = 1;
  string path = 2;  // The file path/location on the orchestrator
}

message FetchFileResponse {
  bytes contents = 1;  // Raw file contents
  string error = 2;    // Error message if fetch failed
}
